src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -148,6 +146,8 @@ public ProNonCombatMoveAI(final ProAI ai) {
           break;
         }
       }
+    } else {
+      moveUnitsToBestTerritories();
```

Comment:
missed all the code expansion blocks in the review, I see the preceding if statement now -  I apologize for the confusion. This code is really rough, looks like the while loop will generally execute only twice.. Shouldn't lines 136 + 144 be at the same log level?

Why move on so quickly from this code rather than fix up stuff around it?